actions/views/cookie.js

```diff
@@ -15,3 +17,11 @@ export function clearUserCookie() {
     document.cookie = `MMCSRF=;expires=Thu, 01 Jan 1970 00:00:01 GMT;domain=${window.location.hostname};path=/`;
     document.cookie = `MMCSRF=;expires=Thu, 01 Jan 1970 00:00:01 GMT;domain=${window.location.hostname};path=${window.basename}`;
 }
+setModule('actions/views/cookie/default', clearUserCookieDefault);
```

Comment:
I assume that this is being done so that the desktop app can call `clearUserCookieDefault` from its overridden version. Is that correct?

If that's just the case, could we instead just have the web app clear whatever extra cookies the desktop app needs as well? The web app calling the desktop app which calls the web app back just feels awkward.

This just makes me wish that we had event bus on the frontend since then the desktop app could just wait for some LOGOUT event and do whatever cleanup it needs then

Reply:
The problem here is the way cookies have to be managed from the Electron side. We manually store them in the view because they need to be manually attached on every request, since Chrome will not a) pass up cookies from an origin other than the current site or b) store cookies at all for an app running off a local static HTML page.

So we have to in addition to what the web app does (to preserve its functionality) we have to manually clear the cookies from the Electron store as well on logout.

Reply:
Ah, I see more of why we're doing this then. So it's not that there's new cookies but rather that `document.cookie` doesn't work the way we need it to?

If that's the case, we're already overwriting `document.cookie` in the desktop app, so could we still have it so that setting the cookies to empty from the web app like we do above causes the desktop app to remove them from the cookie manager?

Although, maybe that's not the case since I wouldn't see why we'd need to also call the default version of `clearUserCookies` from the desktop app then if it doesn't use `document.cookie` in that way

---

On a similar vein, instead of having the web app register a default version of the handler, could we instead make it so that instead of again having the web app overwrite `clearUserCookies` directly, the web app version was more like
```js
export function clearUserCookies() {
    // call whatever the desktop app passed to do when clearing cookies

    document.cookie = ...; // The current stuff
```

I know the naming of the module registry doesn't really match this use case, but all this stuff of changing the const value that's exported based on it still really feels like we're doing things the wrong way.  It's something about the value of a const changing at load time based on something that isn't a compile time flag that feels wrong to me

Reply:
Alright I was able to remove the override by having the desktop app set the cookie manually when the property was updated. Definitely a cleaner approach :)